iceoryx_hoofs/include/iceoryx_hoofs/internal/cxx/storable_function.inl

```diff
@@ -173,14 +195,21 @@ inline void storable_function<Capacity, signature<ReturnType, Args...>>::copy(co
     auto ptr = dest.m_storage.template allocate<CallableType>();
     cxx::Expects(ptr != nullptr);
 
-    auto obj = static_cast<CallableType*>(src.m_callable);
+    // AXIVION Next Construct AutosarC++19_03-M5.2.8: type erasure - conversion to compatible type
+    const auto obj = static_cast<CallableType*>(src.m_callable);
+
+    // AXIVION Next Construct AutosarC++19_03-A5.3.2: obj is guaranteed not to be
```

Comment:
Torn on this rule. Of course we do not want to derefernce a `nullptr`. 
However, this is often guaranteed by invariants, e.g. we know `m_callable` is not null by design (except for moved from objects, which are illegal to access). Axivion does often fails to infer this, which is understandable as it is a non-trivial problem.

Reply:
@MatthiasKillat I disagree here. I think `A5.3.2` is there for a reason. A user could run into the issue of using a moved-from object and then copying it, hence we should check for `nullptr` here with Expects/assert.

Reply:
Yes but then much of the advantage of non-nullability goes right out of the window (being able to assume it is not null).

Note that I do not agree to have this construct not be null in the first place, as it is useful in some designs where we want to store a null function (this better should have been an extension/separate implementation). This is another issue, however.